eda/src/autogluon/eda/analysis/explain.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from typing import List, Optional
+
+import pandas as pd
+import shap
+
+from autogluon.core.constants import REGRESSION
+from autogluon.eda import AnalysisState
+from autogluon.eda.analysis.base import AbstractAnalysis
+
+__all__ = ["ShapAnalysis"]
+
+logger = logging.getLogger(__name__)
+
+
+class _ShapAutogluonWrapper:
+    def __init__(self, predictor, feature_names, target_class=None):
+        self.ag_model = predictor
+        self.feature_names = feature_names
+        self.target_class = target_class
+        if target_class is None and predictor.problem_type != REGRESSION:
+            logging.warning("Since target_class not specified, SHAP will explain predictions for each class")
+
+    def predict_proba(self, X):
+        if isinstance(X, pd.Series):
+            X = X.values.reshape(1, -1)
+        if not isinstance(X, pd.DataFrame):
+            X = pd.DataFrame(X, columns=self.feature_names)
+        if self.ag_model.problem_type == REGRESSION:
+            preds = self.ag_model.predict(X)
+        else:
+            preds = self.ag_model.predict_proba(X)
+        if self.ag_model.problem_type == REGRESSION or self.target_class is None:
+            return preds
+        else:
+            return preds[self.target_class]
```

Comment:
I see a similar hack here that I see in the general AutoGluon Tabular code of using `predict_proba` for regression. Long term do you think this is the right thing to do long-term? is it used to simplify the code logic or just to align with how Tabular does things?